src/mono/wasm/runtime/debug.ts

```diff
@@ -63,7 +65,8 @@ export function mono_wasm_send_dbg_command_with_parms(id: number, command_set: n
     mono_wasm_malloc_and_set_debug_buffer(command_parameters);
     cwraps.mono_wasm_send_dbg_command_with_parms(id, command_set, command, _debugger_buffer, length, valtype, newvalue.toString());
 
-    const { res_ok, res } = commands_received;
+    const { res_ok, res } = commands_received.get(id) as CommandResponse;
+    commands_received.delete(id);
```

Comment:
```suggestion
    const { res_ok, res } = commands_received.get(id) as CommandResponse;
    commands_received.delete(id);
```
I'd suggest making a method that combines get and delete